### PR TITLE
Add unit tests of cloud functionality

### PR DESCRIPTION
**Pull request by mbaynton**
There's still room to add more coverage to these, mostly around different parameters to deployApp on the first deployment, but this is a nice start.

### CODE REVIEW

File: R/deploymentTarget.R [hunk 1]
```diff
@@ -89,14 +89,14 @@ deploymentTargetForApp <- function(appId,
                                    account = NULL,
                                    server = NULL) {
   accountDetails <- findAccount(account, server)
-  application <- getApplication(accountDetails$account, accountDetails$server, appId)
```
Comment (mbaynton) [inline R/deploymentTarget.R:92]: These changes are separately submitted in https://github.com/rstudio/rsconnect/pull/805

File: tests/testthat/test-client-cloud.R [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+
+        list(
+          id=application_id,
+          content_id=application_id-1,
+          name=paste("testthat app", application_id),
+          type=expectedAppType
+        )
+      }
+    ),
+    "^POST /v1/bundles/123/status$" = list(
```
Comment (omar-rs) [inline tests/testthat/test-client-cloud.R:369]: Is this supposed to be a GET instead of a POST?

**Review by omar-rs — COMMENTED** [review]
🎉

**Review by omar-rs — APPROVED** [review]
Looks good to me, with one minor comment.

**Review by mslynch — APPROVED** [review]

File: tests/testthat/test-client-cloud.R (hunk 2 above)
Comment (mbaynton) [reply tests/testthat/test-client-cloud.R:369]: No, this is a real call we expect rsconnect to make. It's like 1. Create bundle, status is pending; 2. Upload bundle to S3; 3. Let lucid-server know we've done that by making the call mocked here.

**Pull request merged**